packages/core/admin/admin/src/content-manager/pages/ListView/components/FieldPicker/index.js

```diff
@@ -27,55 +44,63 @@ export const FieldPicker = ({ layout }) => {
 
   const values = displayedHeaders.map(({ name }) => name);
 
-  const handleChange = (updatedValues) => {
-    trackUsage('didChangeDisplayedFields');
+  const isSelected = (headerName) => values.includes(headerName);
 
-    // removing a header
-    if (updatedValues.length < values.length) {
-      const removedHeader = values.filter((value) => {
-        return updatedValues.indexOf(value) === -1;
-      });
+  const handleChange = (headerName) => {
+    trackUsage('didChangeDisplayedFields');
 
-      dispatch(onChangeListHeaders({ name: removedHeader[0], value: true }));
+    // remove a header
+    if (values.includes(headerName)) {
+      dispatch(onChangeListHeaders({ name: headerName, value: true }));
+      // add a header
     } else {
-      const addedHeader = updatedValues.filter((value) => {
-        return values.indexOf(value) === -1;
-      });
-
-      dispatch(onChangeListHeaders({ name: addedHeader[0], value: false }));
+      dispatch(onChangeListHeaders({ name: headerName, value: false }));
     }
   };
 
+  const handleReset = () => {
+    dispatch(onResetListHeaders());
+  };
+
   return (
-    <Box paddingTop={1} paddingBottom={1}>
-      <Select
-        aria-label="change displayed fields"
-        value={values}
-        onChange={handleChange}
-        customizeContent={(values) =>
-          formatMessage(
-            {
-              id: getTrad('select.currently.selected'),
-              defaultMessage: '{count} currently selected',
-            },
-            { count: values.length }
-          )
-        }
-        multi
-        size="S"
-      >
-        {allAllowedHeaders.map((header) => {
-          return (
-            <Option key={header.name} value={header.name}>
+    <>
```

Comment:
I think you don't need a fragment here.

Reply:
Why not? I have two Flex at the root of the component